### PR TITLE
Story/dfc 2543 repointing to new feed

### PR DESCRIPTION
**Pull request by Gurnaik-Randhawa**
Refactoring to use the new two phase Apprenticeship  API service.

### CODE REVIEW

File: DFC.Integration.AVFeed/DFC.Intergration.AVFeed.Function.GetAVDetailsForProjected/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        /// <summary>
+        /// Runs the specified my queue item.
+        /// </summary>
+        /// <param name="myQueueItem">My queue item.</param>
+        /// <param name="mode">The mode.</param>
+        public static async Task<ProjectedVacancyDetails> RunAsync(ProjectedVacancySummary myQueueItem, RunMode mode, IAsyncCollector<AuditRecord<object, object>> asyncCollector, AuditRecord<object, object> masterRecord)
+        {
+            var container = ConfigureContainer(mode, asyncCollector, masterRecord);
+            var getDetailsFunc = container.Resolve<IGetAvDetailsByIdsFunc>();
+            var auditService = container.Resolve<IAuditService>();
```
Comment (Muthuramana) [inline DFC.Integration.AVFeed/DFC.Intergration.AVFeed.Function.GetAVDetailsForProjected/Startup.cs]: auditService  is never used.

File: DFC.Integration.AVFeed/DFC.Intergration.AVFeed.Function.GetAVDetailsForProjected/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        /// <param name="mode">The mode.</param>
+        /// <returns></returns>
+        public static ILifetimeScope ConfigureContainer(RunMode mode, IAsyncCollector<AuditRecord<object, object>> asyncCollector, AuditRecord<object, object> masterRecord)
+        {
+            var builder = ConfigureDI.ConfigureContainerWithCommonModules(mode);
+            if (mode == RunMode.Azure)
+            {
+                builder.Register(ctx => new CosmosAuditService(asyncCollector, masterRecord));
+            }
+            builder.RegisterType<InMemoryAuditService>().As<IAuditService>().SingleInstance();
```
Comment (Muthuramana) [inline DFC.Integration.AVFeed/DFC.Intergration.AVFeed.Function.GetAVDetailsForProjected/Startup.cs]: I dont think InMemoryAuditService would be a good fit for this function, try looking at DFC.Integration.AVFeed.Function.GetAVForSoc for alternative implementation.

If InMemory must be used, then ensure these audit messages are pumped out to the IAsyncCollector in the AzFunc

File: DFC.Integration.AVFeed/DFC.Integration.AVFeed.AzureFunctions/GetAVDetailsForProjectedFunction.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+using Microsoft.Azure.WebJobs.Host;
+using DFC.Integration.AVFeed.Data.Models;
+using System.Linq;
+using DFC.Integration.AVFeed.Function.Common;
+using DFC.Integration.AVFeed.Core;
+using System.Threading.Tasks;
+
+namespace DFC.Integration.AVFeed.Function.GetAVDetailsForProjected.AzFunc
+{
+    public static class GetAVDetailsForProjectedFunction
```
[the remark this answers is not in the document]
Comment (Gurnaik-Randhawa) [reply DFC.Integration.AVFeed/DFC.Integration.AVFeed.AzureFunctions/GetAVDetailsForProjectedFunction.cs]: Renamed to GetAVDetailsForProjectedAVFunction

File: DFC.Integration.AVFeed/DFC.Integration.AVFeed.Service.AVAPI/AVAPIService.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+            logger.Trace($"Extracting AV summaries for SOC {mapping.SocCode} page : {pageNumber}");
+           
+            var queryString = HttpUtility.ParseQueryString(string.Empty);
+
+            queryString["standardLarsCodes"] = string.Join(",", mapping.Standards);
+            queryString["frameworkLarsCodes"] = string.Join(",", mapping.Frameworks);
+            queryString["pageSize"] = ConfigurationManager.AppSettings.Get("FAA.PageSize");;
+            queryString["pageNumber"] = pageNumber.ToString();
+            queryString["sortBy"] = "Age";
```
[the remark this answers is not in the document]
Comment (Gurnaik-Randhawa) [reply DFC.Integration.AVFeed/DFC.Integration.AVFeed.Service.AVAPI/AVAPIService.cs]: Made configurable , updated config and template files to reflect.

**Pull request merged**